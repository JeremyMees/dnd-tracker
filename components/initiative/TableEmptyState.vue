<script setup lang="ts">
defineProps<{ campaign?: boolean }>()
</script>

<template>
  <UiCard class="border-background bg-background/50">
    <UiCardHeader>
      <UiCardTitle>
        {{ $t('components.initiativeTable.emptyState.title') }}
      </UiCardTitle>
      <UiCardDescription>
        {{ $t('components.initiativeTable.emptyState.text') }}
      </UiCardDescription>
    </UiCardHeader>
    <UiCardContent class="flex flex-col gap-4">
      <div>
        <div class="flex items-center gap-2">
          <Icon
            name="tabler:bat"
            class="size-4 min-w-4 text-destructive"
            aria-hidden="true"
          />
          <span>{{ $t('general.bestiary') }}</span>
        </div>
        <p class="text-muted-foreground">
          {{ $t('components.initiativeTable.emptyState.bestiary') }}
        </p>
      </div>
      <div v-if="campaign">
        <div class="flex items-center gap-2">
          <Icon
            name="tabler:meeple"
            class="size-4 min-w-4 text-primary"
            aria-hidden="true"
          />
          <span>{{ $t('general.campaignHomebrew') }}</span>
        </div>
        <p class="text-muted-foreground">
          {{ $t('components.initiativeTable.emptyState.campaignHomebrew') }}
        </p>
      </div>
      <div>
        <div class="flex items-center gap-2">
          <Icon
            name="tabler:beer"
            class="size-4 min-w-4 text-warning"
            aria-hidden="true"
          />
          <span>{{ $t('general.newHomebrew') }}</span>
        </div>
        <p class="text-muted-foreground">
          {{ $t('components.initiativeTable.emptyState.newHomebrew') }}
        </p>
      </div>
    </UiCardContent>
  </UiCard>
</template>
