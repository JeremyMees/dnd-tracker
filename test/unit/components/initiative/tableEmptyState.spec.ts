import { mountSuspended } from '@nuxt/test-utils/runtime'
import { describe, expect, it, vi } from 'vitest'
import TableEmptyState from '~/components/initiative/TableEmptyState.vue'

describe('Initiative table empty state', () => {
  it('Should match snapshot with campaign prop', async () => {
    const component = await mountSuspended(TableEmptyState, { props: { campaign: true } })

    expect(component.html()).toMatchSnapshot()
  })

  it('Should match snapshot without campaign prop', async () => {
    const component = await mountSuspended(TableEmptyState, { props: { campaign: false } })

    expect(component.html()).toMatchSnapshot()
  })
})
