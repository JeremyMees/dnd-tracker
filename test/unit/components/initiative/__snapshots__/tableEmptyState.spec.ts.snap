// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Initiative table empty state > Should match snapshot with campaign prop 1`] = `
"<div class="rounded-lg border-4 text-card-foreground shadow-sm border-background bg-background/50">
  <div class="flex flex-col gap-y-1.5 p-6">
    <h3 class="text-2xl font-semibold leading-none tracking-tight">components.initiativeTable.emptyState.title</h3>
    <p class="text-sm text-muted-foreground">components.initiativeTable.emptyState.text</p>
  </div>
  <div class="p-6 pt-0 flex flex-col gap-4">
    <div>
      <div class="flex items-center gap-2"><span class="iconify i-tabler:bat size-4 min-w-4 text-destructive" aria-hidden="true"></span><span>general.bestiary</span></div>
      <p class="text-muted-foreground">components.initiativeTable.emptyState.bestiary</p>
    </div>
    <div data-test-campaign="">
      <div class="flex items-center gap-2"><span class="iconify i-tabler:meeple size-4 min-w-4 text-primary" aria-hidden="true"></span><span>general.campaignHomebrew</span></div>
      <p class="text-muted-foreground">components.initiativeTable.emptyState.campaignHomebrew</p>
    </div>
    <div>
      <div class="flex items-center gap-2"><span class="iconify i-tabler:beer size-4 min-w-4 text-warning" aria-hidden="true"></span><span>general.newHomebrew</span></div>
      <p class="text-muted-foreground">components.initiativeTable.emptyState.newHomebrew</p>
    </div>
  </div>
</div>"
`;

exports[`Initiative table empty state > Should match snapshot without campaign prop 1`] = `
"<div class="rounded-lg border-4 text-card-foreground shadow-sm border-background bg-background/50">
  <div class="flex flex-col gap-y-1.5 p-6">
    <h3 class="text-2xl font-semibold leading-none tracking-tight">components.initiativeTable.emptyState.title</h3>
    <p class="text-sm text-muted-foreground">components.initiativeTable.emptyState.text</p>
  </div>
  <div class="p-6 pt-0 flex flex-col gap-4">
    <div>
      <div class="flex items-center gap-2"><span class="iconify i-tabler:bat size-4 min-w-4 text-destructive" aria-hidden="true"></span><span>general.bestiary</span></div>
      <p class="text-muted-foreground">components.initiativeTable.emptyState.bestiary</p>
    </div>
    <!--v-if-->
    <div>
      <div class="flex items-center gap-2"><span class="iconify i-tabler:beer size-4 min-w-4 text-warning" aria-hidden="true"></span><span>general.newHomebrew</span></div>
      <p class="text-muted-foreground">components.initiativeTable.emptyState.newHomebrew</p>
    </div>
  </div>
</div>"
`;
